### PR TITLE
LibJS: Stub out Atomics.wait, Atomics.waitAsync, and Atomics.notify

### PR DESCRIPTION
**Pull request by trflynn89**
```
Diff Tests:
    +58 ✅    -85 ❌   +2 ⚠️   +25 📝   

Diff Tests:
    test/built-ins/Atomics/notify/bad-range.js                                                ❌ -> ✅
    test/built-ins/Atomics/notify/bigint/bad-range.js                                         ❌ -> ✅
    test/built-ins/Atomics/notify/bigint/non-shared-bufferdata-count-evaluation-throws.js     ❌ -> ✅
    test/built-ins/Atomics/notify/bigint/non-shared-bufferdata-index-evaluation-throws.js     ❌ -> ✅
    test/built-ins/Atomics/notify/bigint/non-shared-bufferdata-returns-0.js                   ❌ -> ✅
    test/built-ins/Atomics/notify/count-boundary-cases.js                                     ❌ -> 📝
    test/built-ins/Atomics/notify/count-from-nans.js                                          ❌ -> 📝
    test/built-ins/Atomics/notify/count-tointeger-throws-then-wake-throws.js                  ❌ -> ✅
    test/built-ins/Atomics/notify/descriptor.js                                               ❌ -> ✅
    test/built-ins/Atomics/notify/length.js                                                   ❌ -> ✅
    test/built-ins/Atomics/notify/name.js                                                     ❌ -> ✅
    test/built-ins/Atomics/notify/negative-index-throws.js                                    ❌ -> ✅
    test/built-ins/Atomics/notify/non-shared-bufferdata-count-evaluation-throws.js            ❌ -> ✅
    test/built-ins/Atomics/notify/non-shared-bufferdata-index-evaluation-throws.js            ❌ -> ✅
    test/built-ins/Atomics/notify/non-shared-bufferdata-returns-0.js                          ❌ -> ✅
    test/built-ins/Atomics/notify/not-a-constructor.js                                        ❌ -> ✅
    test/built-ins/Atomics/notify/out-of-range-index-throws.js                                ❌ -> ✅
    test/built-ins/Atomics/notify/symbol-for-index-throws.js                                  ❌ -> ✅
    test/built-ins/Atomics/wait/bad-range.js                                                  ❌ -> ✅
    test/built-ins/Atomics/wait/bigint/bad-range.js                                           ❌ -> ✅
    test/built-ins/Atomics/wait/bigint/cannot-suspend-throws.js                               ✅ -> ⚠️
    test/built-ins/Atomics/wait/bigint/false-for-timeout.js                                   ❌ -> 📝
    test/built-ins/Atomics/wait/bigint/negative-index-throws.js                               ❌ -> ✅
    test/built-ins/Atomics/wait/bigint/negative-timeout.js                                    ❌ -> 📝
    test/built-ins/Atomics/wait/bigint/out-of-range-index-throws.js                           ❌ -> ✅
    test/built-ins/Atomics/wait/cannot-suspend-throws.js                                      ✅ -> ⚠️
    test/built-ins/Atomics/wait/descriptor.js                                                 ❌ -> ✅
    test/built-ins/Atomics/wait/false-for-timeout.js                                          ❌ -> 📝
    test/built-ins/Atomics/wait/length.js                                                     ❌ -> ✅
    test/built-ins/Atomics/wait/name.js                                                       ❌ -> ✅
    test/built-ins/Atomics/wait/negative-index-throws.js                                      ❌ -> ✅
    test/built-ins/Atomics/wait/negative-timeout.js                                           ❌ -> 📝
    test/built-ins/Atomics/wait/null-for-timeout.js                                           ❌ -> 📝
    test/built-ins/Atomics/wait/object-for-timeout.js                                         ❌ -> 📝
    test/built-ins/Atomics/wait/out-of-range-index-throws.js                                  ❌ -> ✅
    test/built-ins/Atomics/wait/poisoned-object-for-timeout-throws.js                         ❌ -> ✅
    test/built-ins/Atomics/wait/symbol-for-index-throws.js                                    ❌ -> ✅
    test/built-ins/Atomics/wait/symbol-for-timeout-throws.js                                  ❌ -> ✅
    test/built-ins/Atomics/wait/symbol-for-value-throws.js                                    ❌ -> ✅
    test/built-ins/Atomics/wait/true-for-timeout.js                                           ❌ -> 📝
    test/built-ins/Atomics/waitAsync/bad-range.js                                             ❌ -> ✅
    test/built-ins/Atomics/waitAsync/bigint/bad-range.js                                      ❌ -> ✅
    test/built-ins/Atomics/waitAsync/bigint/false-for-timeout.js                              ❌ -> 📝
    test/built-ins/Atomics/waitAsync/bigint/negative-index-throws.js                          ❌ -> ✅
    test/built-ins/Atomics/waitAsync/bigint/negative-timeout.js                               ❌ -> 📝
    test/built-ins/Atomics/waitAsync/bigint/non-bigint64-typedarray-throws.js                 ❌ -> ✅
    test/built-ins/Atomics/waitAsync/bigint/non-shared-bufferdata-throws.js                   ❌ -> ✅
    test/built-ins/Atomics/waitAsync/bigint/not-a-typedarray-throws.js                        ❌ -> ✅
    test/built-ins/Atomics/waitAsync/bigint/not-an-object-throws.js                           ❌ -> ✅
    test/built-ins/Atomics/waitAsync/bigint/null-bufferdata-throws.js                         ❌ -> ✅
    test/built-ins/Atomics/waitAsync/bigint/null-for-timeout.js                               ❌ -> 📝
    test/built-ins/Atomics/waitAsync/bigint/object-for-timeout.js                             ❌ -> 📝
    test/built-ins/Atomics/waitAsync/bigint/out-of-range-index-throws.js                      ❌ -> ✅
    test/built-ins/Atomics/waitAsync/bigint/poisoned-object-for-timeout-throws.js             ❌ -> ✅
    test/built-ins/Atomics/waitAsync/bigint/symbol-for-index-throws.js                        ❌ -> ✅
    test/built-ins/Atomics/waitAsync/bigint/symbol-for-timeout-throws.js                      ❌ -> ✅
    test/built-ins/Atomics/waitAsync/bigint/symbol-for-value-throws.js                        ❌ -> ✅
    test/built-ins/Atomics/waitAsync/bigint/undefined-for-timeout.js                          ❌ -> 📝
    test/built-ins/Atomics/waitAsync/bigint/value-not-equal.js                                ❌ -> 📝
    test/built-ins/Atomics/waitAsync/descriptor.js                                            ❌ -> ✅
    test/built-ins/Atomics/waitAsync/false-for-timeout.js                                     ❌ -> 📝
    test/built-ins/Atomics/waitAsync/implicit-infinity-for-timeout.js                         ❌ -> 📝
    test/built-ins/Atomics/waitAsync/is-function.js                                           ❌ -> ✅
    test/built-ins/Atomics/waitAsync/length.js                                                ❌ -> ✅
    test/built-ins/Atomics/waitAsync/name.js                                                  ❌ -> ✅
    test/built-ins/Atomics/waitAsync/negative-index-throws.js                                 ❌ -> ✅
    test/built-ins/Atomics/waitAsync/negative-timeout.js                                      ❌ -> 📝
    test/built-ins/Atomics/waitAsync/non-int32-typedarray-throws.js                           ❌ -> ✅
    test/built-ins/Atomics/waitAsync/non-shared-bufferdata-throws.js                          ❌ -> ✅
    test/built-ins/Atomics/waitAsync/not-a-typedarray-throws.js                               ❌ -> ✅
    test/built-ins/Atomics/waitAsync/not-an-object-throws.js                                  ❌ -> ✅
    test/built-ins/Atomics/waitAsync/null-bufferdata-throws.js                                ❌ -> ✅
    test/built-ins/Atomics/waitAsync/null-for-timeout.js                                      ❌ -> 📝
    test/built-ins/Atomics/waitAsync/object-for-timeout.js                                    ❌ -> 📝
    test/built-ins/Atomics/waitAsync/out-of-range-index-throws.js                             ❌ -> ✅
    test/built-ins/Atomics/waitAsync/poisoned-object-for-timeout-throws.js                    ❌ -> ✅
    test/built-ins/Atomics/waitAsync/returns-result-object-value-is-promise-resolves-to-ok.js ❌ -> 📝
    test/built-ins/Atomics/waitAsync/returns-result-object-value-is-string-not-equal.js       ❌ -> 📝
    test/built-ins/Atomics/waitAsync/returns-result-object-value-is-string-timed-out.js       ❌ -> 📝
    test/built-ins/Atomics/waitAsync/symbol-for-index-throws.js                               ❌ -> ✅
    test/built-ins/Atomics/waitAsync/symbol-for-timeout-throws.js                             ❌ -> ✅
    test/built-ins/Atomics/waitAsync/symbol-for-value-throws.js                               ❌ -> ✅
    test/built-ins/Atomics/waitAsync/undefined-for-timeout.js                                 ❌ -> 📝
    test/built-ins/Atomics/waitAsync/validate-arraytype-before-index-coercion.js              ❌ -> ✅
    test/built-ins/Atomics/waitAsync/validate-arraytype-before-timeout-coercion.js            ❌ -> ✅
    test/built-ins/Atomics/waitAsync/validate-arraytype-before-value-coercion.js              ❌ -> ✅
    test/built-ins/Atomics/waitAsync/value-not-equal.js                                       ❌ -> 📝
```

### CODE REVIEW

File: Userland/Libraries/LibJS/Runtime/AtomicsObject.cpp [hunk 1]
```diff
@@ -401,6 +461,75 @@ JS_DEFINE_NATIVE_FUNCTION(AtomicsObject::sub)
...
+
+    // FIXME: Implement the remaining steps when we support SharedArrayBuffer.
+    (void)byte_index_in_buffer;
+    (void)count;
+    (void)block;
+
+    return vm.throw_completion<InternalError>(ErrorType::NotImplemented, "SharedArrayBuffer"sv);
+}
+
 // 25.4.14 Atomics.xor ( typedArray, index, value ), https://tc39.es/ecma262/#sec-atomics.xor
```
Comment (shannonbooth) [inline Userland/Libraries/LibJS/Runtime/AtomicsObject.cpp:533]: Should this be `25.4.16` now?

**Review by shannonbooth — APPROVED** [review]
looks good to me!

File: Userland/Libraries/LibJS/Runtime/AtomicsObject.cpp (hunk 1 above)
Comment (trflynn89) [reply Userland/Libraries/LibJS/Runtime/AtomicsObject.cpp:533]: It should, there's a bunch numbers that we're missing - I have another branch where I've gone through and updated them all, so I left them alone here

Comment (shannonbooth) [reply Userland/Libraries/LibJS/Runtime/AtomicsObject.cpp:533]: Fair enough :^)

**Review by awesomekling — APPROVED** [review]

**Pull request merged**